File: coroutine.js
function* countdown(start) {
  var val = start;
  while (val >= 0) {
    var delta = (yield val) || 1;
    val -= delta
  }
}

generator_object = countdown(10);
print = console.log;
print(generator_object.next());
print(generator_object.next());
print(generator_object.next(5)); // instead of send(5)
print(generator_object.next());
try {
  generator_object.throw(new Error('foo')) // has no thrilling effect on our generator since it doesnt try-catch it
}
catch (ex) {
  print("generator re-threw exception: ", ex.message);
}

File: coroutine.py
def countdown(start):
  val = start
  while val >= 0:
    delta = (yield val) or 1
    val -= delta

generator_object = countdown(10)
print(generator_object.__next__()) # aka next(generator_object)
print(generator_object.__next__())
print(generator_object.send(5))
print(generator_object.__next__())
try:
  generator_object.throw(Exception('foo')) # has no thrilling effect on our generator since it doesnt try-catch it
except:
  print("generator re-threw exception")